### ISSUE TITLE
sunxi-v3s uart1's reg can't operate

### ISSUE THREAD

**Issue by DongStop** [linux-sunxi/linux-sunxi#302]
When I probe uart1 driver written by myself,I do remap the regs(START:0x1C28400 OFFSET:0x400).However,All the regs are zero.The picture  as follows
![image](https://user-images.githubusercontent.com/66958826/162887785-1aaaa278-a803-4dc4-9553-2043b42658d7.png)
It is ok when I use the driver to read uart0's reg. BUS_CLK_GATING_REG3:0x30001.I think the uart1 clock is ready.Could you tell me how to solve this problem.

**Comment by wens** [comment]
Is the UART out of reset?

**Comment by DongStop** [comment]
> Is the UART out of reset?

yes,In uboot it is no need to use the UART1 port.In that case I ignore the UART1 peripheral at that time.the UART peripheral is soc periphral.Does it reset following the CPU's booting process?Or are there some other ways to reset the soc peripheral through register?I already try to reset the UART1 through clock.when the driver didn't be loaded the BUS_CLK_GATING_REG3's value is 0x10001.then do 'insmod' the driver,the value turns to 0x30001.It still didn't work.the reg's information as follows

![1](https://user-images.githubusercontent.com/66958826/163092580-e5190702-1310-409d-b4b0-4e2d925969cd.PNG)
![2](https://user-images.githubusercontent.com/66958826/163092617-1c460e36-516d-437a-90fa-cc848fdc1eb1.PNG)
